### PR TITLE
Improve Transport APIs

### PR DESCRIPTION
**Pull request by f3rno**
* add query args field to REST2 `candles()` query
 * allow duplicate candle/OB snapshots
 * allow filtering `o*` event packets by all IDs & symbol

### CODE REVIEW

File: lib/transports/rest2.js [hunk 1]
```diff
@@ -15,12 +15,24 @@ const {
...
 
+// TODO: Extract into util/use in RESTv1?
+const integrateQuery = (url, query) => {
+  const params = Object.keys(query)
+
+  if (params.length > 0) {
+    return `${url}?${params.map(p => `${p}=${query[p]}`).join('&')}`
+  }
+
+  return url
```
Comment (robertkowalski) [inline lib/transports/rest2.js]: take a look here: https://nodejs.org/api/url.html#url_class_urlsearchparams

File: lib/transports/ws2.js [hunk 2]
```diff
@@ -465,11 +465,9 @@ class WSv2 extends EventEmitter {
...
    */
   _updateManagedOB (symbol, data) {
-    if (Array.isArray(data[0])) { // snapshot, new OB
-      if (this._orderBooks[symbol]) {
-        return new Error(`recv snapshot for known OB: ${symbol}`)
-      }
-
+    // Snapshot, new OB. Note that we don't protect against duplicates, as they
+    // could come in on re-sub
+    if (Array.isArray(data[0])) {
```
Comment (robertkowalski) [inline lib/transports/ws2.js]: its not clear to me what that fixes?

maybe write it int he commit message or test?

Comment (robertkowalski) [reply lib/transports/ws2.js]: hmmm... what was the protection against duplicates used for?

Comment (f3rno) [reply lib/transports/ws2.js]: It was a mistake I made initially, I assumed the snapshots only ever arrive once per-connection, not once per-subscription. With the old behavior an error would be thrown if a duplicate snapshot arrived (meant as a sanity check.)

File: lib/transports/rest2.js (hunk 1 above)
Comment (f3rno) [reply lib/transports/rest2.js]: TIL :slightly_smiling_face: fixed

**Comment by robertkowalski** [comment]
very cool! thx!

**Comment by robertkowalski** [comment]
👍

**Pull request merged**